pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -704,6 +755,21 @@ func (r *ReconcileInfinispan) upgradeInfinispan(infinispan *infinispanv1.Infinis
 	})
 }
 
+func (r *ReconcileInfinispan) dropSecretOwnerReference(secretName string, infinispan *infinispanv1.Infinispan, logger logr.Logger) error {
+	secret := &corev1.Secret{}
+	if result, err := kube.LookupResource(secretName, infinispan.Namespace, secret, r.client, logger, eventRec); result != nil {
+		return err
+	}
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, secret, func() error {
```

Comment:
Do we need todo `LookupResource` and `CreateOrUpdate`? Can't we just use `CreateOrPatch`?

The value of `result` is never used, as eventually `destroyResources` always returns `reconcile.Result{}, err` if an error exists.

Reply:
Yes, good point for optimization. Thanks, @ryanemerson